### PR TITLE
Added anchor tags to the html. Fixes #28

### LINKED ISSUE #28: Alow fragment links to categories in categories.html
Alow links to categories such as `https://mulberrysymbols.org/assets/categories/categories.html#transport-space`. Will obviously be slow links as v large file

* Add <a> tags with id to act as fragment anchor
* slugify category

### PR DESCRIPTION
**Pull request by gavinhenderson**
Added anchor tags to the html so you can now link to things by doing #category-here

Addressing #28

### CODE REVIEW

File: scripts/mkhtml.js [hunk 1]
```diff
@@ -79,6 +81,14 @@ async function asyncGenerateHtmlContent(data) {
...
   const htmlTemplate = handlebars.compile(htmlFile);
 
+  Object.keys(sortedCategories).forEach((currentKey) => {
+    const symbolsArr = sortedCategories[currentKey];
+    sortedCategories[currentKey] = {
+      anchor: slugify(currentKey),
+      symbols: symbolsArr,
+    };
+  });
+
```
Comment (SteveALee) [inline scripts/mkhtml.js]: Personally I'd avoid mutating the sortedCategories and create a new one. This unmutable style can save lots of bugs and confusion when debugging. One way to do it is as in lines 70-74 above.

However, there is a subtle bug in the use of `Object.keys(sortedCategories)` here. The resulting array is officially in arbitrary order. Hence you need to call .sort() again, as in line 72. In reality, node almost certainly iterates in sequence as is being assumed, so everything works as expected. Or perhaps node (or rather V8) specifies the iteration order, I don't know.

I'm now worried about the same bug applying when the template is applied. Unless handlebars handles `#each` iteraction specially I excpect the same potenial bug applies - the [handlebars documentaion doesn't say](ttps://handlebarsjs.com/builtin_helpers.html#iteration].

Ideally, the sortedCategories and the data passed to the template would be an Array. (not sure about Map sorting).

**Comment by SteveALee** [comment]
Actually can I add a bit of feature creep? :)
Can we have a perma link just after the category text. Like [these in WCAG](https://www.w3.org/TR/WCAG21/#later-versions-of-accessibility-guidelines) but not right justified?

There is also a tooltip in those which might help people understand.

```
<span class="permalink">
  <a href="#later-versions-of-accessibility-guidelines" aria-label="Permalink for 0.6 Later Versions of Accessibility Guidelines" title="Permalink for 0.6 Later Versions of Accessibility Guidelines">
   <span>§</span>
  </a>
</span>
```

File: scripts/mkhtml.js (hunk 1 above)
Comment (SteveALee) [reply scripts/mkhtml.js]: Oh and mutating the thing you are iterating is even more dodgy :)

**Comment by gavinhenderson** [comment]
@SteveALee Thanks for the feedback! It felt a bit dodgy when I wrote it so glad you held me accountable to my dodgy code. I am no longer mutating it in place as you suggested. Does my commit solve your concerns?

**Comment by SteveALee** [comment]
Perfect!

Do you fancy doing the [permalink](https://github.com/mulberrysymbols/mulberry-symbols/pull/33#issuecomment-463167192) as well? Or I'm happy to just merge this and make that another issue.

**Comment by gavinhenderson** [comment]
@SteveALee Makes sense just to do it in this PR. Thats the permalink added, whats your thoughts on the styling?

**Comment by SteveALee** [comment]
Minor comments - 
* The link seems a bit to low - should sit on the heading letters baseline
* Can it be less bolder or lighter colour?

**Comment by SteveALee** [comment]
PS the link is also appears in the PDF but that's as mkcategories is shared.
Really not a problem - just observation. :) Refactor would be expensive

**Comment by gavinhenderson** [comment]
Changes made :+1: 

Yeah it would be either an expensive refactor or a really bad fix if it was done quick. At least links definitely work when exported to PDF so the contents page will work on the PDF which is a bonus

**Comment by SteveALee** [comment]
@gavinhenderson  Looks great! Thanks for updating the built files too.

**Pull request merged**